config-example.yaml

```diff
@@ -224,54 +224,67 @@ policy:
 # - https://tailscale.com/kb/1081/magicdns/
 # - https://tailscale.com/blog/2021-09-private-dns-with-magicdns/
 #
-dns_config:
-  # Whether to prefer using Headscale provided DNS or use local.
-  override_local_dns: true
+# Please not that for the DNS configuration to have any effect,
+# clients must have the `--accept-ds=true` option enabled. This is the
+# default for the Tailscale client. This option is enabled by default
+# in the Tailscale client.
+#
+# Setting _any_ of the configuration and `--accept-dns=true` on the
+# clients will integrate with the DNS manager on the client or
+# overwrite /etc/resolv.conf.
+# https://tailscale.com/kb/1235/resolv-conf
+#
+# If you want stop Headscale from managing the DNS configuration
+# all the fields under `dns` should be set to empty values.
+dns:
+  # Whether to use [MagicDNS](https://tailscale.com/kb/1081/magicdns/).
+  # Only works if there is at least a nameserver defined.
+  magic_dns: true
+
+  # Defines the base domain to create the hostnames for MagicDNS.
+  # This domain _must_ be different from the server_url domain.
```

Comment:
Different as in different subdomains or different TLDs?

Would headscale.example.com and scale.example.com work?

Reply:
different (sub)domains, your example works.

Could also do 
headscale.net as the server
clients.headscale.net as clients


Technically could do 
server.headscale.net as the server
headscale.net as clients (as long as you dont have a node called server)

Reply:
we dont need any changes for Nginx proxy Manger also ? 
if was : 
`server_url: https://headscale.site.com `
`base_domain: headscale.site.com`

and will be : 

`server_url: https://headscale.site.com `
`base_domain: clients.headscale.site.com`

Reply:
I dont think so, but I dont know anything about nginx proxy manager.

Reply:
This is a real shame - is there a technical reason why this can't be allowed, or is it a case of being too difficult/risky to support over the rewrite?

I guess what I'm really asking is: is there a possibility of me (or someone else) making a patch to change this down the line?

Reply:
No problem, It is one of these things that "we dont want to add this", but it is how upstream works, and we dont really have any choice, and I understand why it happens upstream too. We just initially implemented it wrongly.